### PR TITLE
[TST] Test sanitize_label()

### PR DESCRIPTION
**Pull request by l-espana**
Hoping this keeps previous code and commits!

### CODE REVIEW

File: heudiconv/tests/test_bids.py [hunk 1]
```diff
@@ -23,4 +24,10 @@ def test_treat_age():
     assert treat_age('1M') == '0.08'
     assert treat_age('12M') == '1'
     assert treat_age('0000.1') == '0.1'
-    assert treat_age(0000.1) == '0.1'
\ No newline at end of file
+    assert treat_age(0000.1) == '0.1'
+
+def sanitize_label():
```
Comment (yarikoptic) [inline heudiconv/tests/test_bids.py:29]:
```suggestion
def test_sanitize_label():
```

**Review by yarikoptic — CHANGES_REQUESTED** [review]
This is much better!  

- left instructions on testing
- since PR is old, conflicts with master piled up, this branch should merge our `master` and resolve conflicts (or rebase -- could be a bit more "involved").  if need help -- I can do that after testing fixed up etc.

**Comment by yarikoptic** [comment]
I still have no clue what is wrong with travis CI, and why appveyor kicked in (we don't use it here AFAIK)... later

File: heudiconv/bids.py [hunk 2]
```diff
@@ -511,10 +510,31 @@ def convert_sid_bids(subject_id):
...
+    label : string
+        Original label
+    """
     cleaner = lambda y: ''.join([x for x in y if x.isalnum()])
-    sid = cleaner(subject_id)
-    if not sid:
+    clean_label = cleaner(label)
+    if not clean_label:
         raise ValueError(
+<<<<<<< HEAD
```
Comment (yarikoptic) [inline heudiconv/bids.py:537]: remove conflict markers/resolve conflicts here and elsewhere

File: heudiconv/bids.py [hunk 3]
```diff
@@ -1024,3 +1044,11 @@ def suffix(self):
...
     def extension(self):
         return self._extension
+=======
+            "Label became empty after cleanup.  Please provide manually "
+            "a suitable alphanumeric label.")
+    if clean_label != label:
+        lgr.warning('{0} contained nonalphanumeric character(s), label '
+                    'was cleaned to be {1}'.format(label, clean_label))
+    return clean_label, label
+>>>>>>> refactor convert_sid_bids to sanitize_label
```
Comment (yarikoptic) [inline heudiconv/bids.py:1054]: smells like some incorrectly resolved conflict

File: heudiconv/tests/test_bids.py [hunk 4]
```diff
@@ -1154,4 +1159,14 @@ def test_ME_mag_phase_conversion(tmpdir, subID='MEGRE', heuristic='bids_ME.py'):
...
-
+=======
+def test_sanitize_label():
+    assert sanitize_label("12345A") == "12345A"
+    assert sanitize_label("12345_A") == "12345A"
+    
+    #non-alphanumerics should raise error
+    with pytest.raises(ValueError):
+        assert sanitize_label(12345)
+        #this should also raise an error for an empty value
```
[the remark this answers is not in the document]
Comment (l-espana) [reply heudiconv/tests/test_bids.py:1170]: Oh, I think I see what you mean. Will update.

**Pull request closed without merging**